crates/containerd-shim-wasm/src/sandbox/shim.rs

```diff
@@ -1,1274 +0,0 @@
-        debug!("wait: {:?}", req);
-        let resp = self.task_wait(req)?;
-        Ok(resp)
-    }
-
-    fn connect(
-        &self,
-        _ctx: &TtrpcContext,
-        req: api::ConnectRequest,
-    ) -> TtrpcResult<api::ConnectResponse> {
-        debug!("connect: {:?}", req);
-
-        let i = self.get_instance(req.id())?;
-        let pid = *i.pid.read().unwrap().as_ref().unwrap_or(&0);
-
-        Ok(api::ConnectResponse {
-            shim_pid: std::process::id(),
-            task_pid: pid,
-            ..Default::default()
-        })
-    }
-
-    fn state(
-        &self,
-        _ctx: &TtrpcContext,
-        req: api::StateRequest,
-    ) -> TtrpcResult<api::StateResponse> {
-        debug!("state: {:?}", req);
-        let resp = self.task_state(req)?;
-        Ok(resp)
-    }
-
-    fn shutdown(&self, _ctx: &TtrpcContext, _req: api::ShutdownRequest) -> TtrpcResult<api::Empty> {
-        debug!("shutdown");
-
-        if self.is_empty().not() {
-            return Ok(api::Empty::new());
-        }
-
-        self.exit.signal();
-
-        Ok(api::Empty::new())
-    }
-
-    fn stats(
-        &self,
-        _ctx: &::ttrpc::TtrpcContext,
-        req: StatsRequest,
-    ) -> ::ttrpc::Result<StatsResponse> {
-        log::info!("stats: {:?}", req);
-        let resp = self.task_stats(req)?;
-        Ok(resp)
-    }
-}
-
-/// Cli implements the containerd-shim cli interface using `Local<T>` as the task service.
-pub struct Cli<T: Instance + Sync + Send> {
-    pub engine: T::Engine,
-    namespace: String,
-    containerd_address: String,
-    phantom: std::marker::PhantomData<T>,
-    exit: Arc<ExitSignal>,
-    _id: String,
-}
-
-impl<I> shim::Shim for Cli<I>
-where
-    I: Instance + Sync + Send,
-    <I as Instance>::Engine: Default,
-{
-    type T = Local<I>;
-
-    fn new(_runtime_id: &str, args: &Flags, _config: &mut shim::Config) -> Self {
-        Cli {
-            engine: Default::default(),
-            phantom: std::marker::PhantomData,
-            namespace: args.namespace.to_string(),
-            containerd_address: args.address.clone(),
-            exit: Arc::new(ExitSignal::default()),
-            _id: args.id.to_string(),
-        }
-    }
-
-    fn start_shim(&mut self, opts: containerd_shim::StartOpts) -> shim::Result<String> {
-        let dir = current_dir().map_err(|err| ShimError::Other(err.to_string()))?;
-        let spec = Spec::load(dir.join("config.json").to_str().unwrap()).map_err(|err| {
-            shim::Error::InvalidArgument(format!("error loading runtime spec: {}", err))
-        })?;
-
-        let default = HashMap::new() as HashMap<String, String>;
-        let annotations = spec.annotations().as_ref().unwrap_or(&default);
-
-        let id = opts.id.clone();
-
-        let grouping = annotations
-            .get("io.kubernetes.cri.sandbox-id")
-            .unwrap_or(&id)
-            .to_string();
-
-        #[cfg(unix)]
-        mount::<str, Path, str, str>(
```

Comment:
Hmm, good catch.
IIRC, the `mount` is not needed because libcontainer takes care of this for us.
I think we don't need the `setup_namespace` either, will try without it.

Reply:
The `setup_namespace` is the one that was removed in #327 [here](https://github.com/containerd/runwasi/pull/327/files#diff-8f8704e2bb5644be97cfbd99c736cb7f432da2de20292de2b0f69e4193bd0f64L1374), which I didn't notice while rebasing.

The `mount` is handled in `libcontainer` [here](https://github.com/containers/youki/blob/710cb4c636bc693a429aea32f840260f0d2b03b2/crates/libcontainer/src/rootfs/rootfs.rs#L55).